libursa/QueryGraph.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <stdint.h>
+
+#include <string>
+
+#include "Core.h"
+
+// Strongly typed integer value, represents a node ID in a QueryGraph.
+class NodeId {
+    uint32_t id_;
+
+   public:
+    NodeId(uint32_t id) : id_(id) {}
+
+    bool operator<(const NodeId &rhs) const { return id_ < rhs.id_; }
+
+    uint32_t get() const { return id_; }
+};
+
+using Edge = std::tuple<NodeId, NodeId>;
```

Comment:
wouldn't std::pair be more appropriate? :thinking:

Reply:
yes it would. I've confused python with C++ :thinking: